discord/poll.py

```diff
@@ -0,0 +1,215 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+
+from typing import (
+    Dict,
+    Optional,
+    List,
+    TYPE_CHECKING,
+    Literal,
+    Union,
+)
+
+from .enums import PollLayoutType, try_enum
+from . import utils
+from .emoji import PartialEmoji, Emoji
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .types.poll import (
+        Poll as PollPayload,
+        _PollAnswer as PollAnswerPayload,
+        _PollAnswerMedia as PollAnswerMediaPayload
+    )
+
+__all__ = (
+    'Poll',
+)
+
+MISSING = utils.MISSING
+
+PollDuration = Literal[
+    1, 24, 72, 168
+]
+
+
+class PollAnswer:
+    """Represents a poll's answer.
+
+    .. container:: operations
+
+        .. describe:: str(x)
+
+            Returns this answer's text, if any.
+    
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The answers display text, or ``None`` if there
+        isn't.
+    attachments: Optional[List[:class:`int`]]
+        The attachment IDs for this answer.
+    """
+
+    __slots__ = (
+        'text',
+        'emoji'
+    )
+
+    def __init__(self, text: str, *, emoji: Union[PartialEmoji, Emoji, str] = MISSING) -> None:
+        self.text: str = text
+        self.emoji: Optional[Union[PartialEmoji, Emoji, str]] = emoji if emoji is not MISSING else None
+
+    @classmethod
+    def _from_data(cls, data: PollAnswerPayload) -> Self:
+        media = data.get('poll_media')
+        text = media.get('text')
+        emoji = media.get('emoji', None)
```

Comment:
```suggestion
        media = data['poll_media']
```
if media is None other stuff breaks anyway.

Reply:
Yeah, you are right, I committed this change (with some others that are pretty similar like this).